src/infoview.ts

```diff
@@ -5,230 +5,212 @@ import {
+                if (!compareMessages(msgs[i], old_msgs[i])) {
+                    eq = false;
+                    break;
                 }
-            });
-        } else {
-            this.curGoal = null;
-            commands.executeCommand('_workbench.htmlPreview.postMessage',
-                Uri.parse('lean-info:goals'),
-                {
-                    command: 'clear-goal'
-                });
+            }
+            if (eq) return;
         }
+        this.curMessages = msgs;
     }
 
-    private renderGoal() {
-        const curGoal = this.curGoal;
-        if (!curGoal) return ``;
-        const f = curGoal.file; const l = curGoal.line; const c = curGoal.column;
-        return `<div id="goal">
-                <h1>Current Goal at ${basename(f)}:${l}:${c}</a></h1>
-                <pre>${curGoal.state}</pre>
-            </div>`;
-    }
+    private updateGoal() {
+        const f = this.curFileName;
+        const l = this.curPosition.line + 1;
+        const c = this.curPosition.character;
+        this.server.info(f, l, c).then((info) => {
+            if (info.record && info.record.state) {
+                if (this.curGoalState !== info.record.state) {
+                    this.curGoalState = info.record.state;
+                    this.fire();
+                }
+            } else {
+                if (this.curGoalState) {
+                    this.curGoalState = null;
+                    this.fire();
+                }
+            }
+        })
 
-    private renderMessages() {
-        if (!this.fileName || !this.showMessages) return ``;
-        const msgs = this.server.messages
-            .filter((m) => m.file_name === this.fileName)
-            .sort((a, b) => a.pos_line === b.pos_line
-                ? a.pos_col - b.pos_col
-                : a.pos_line - b.pos_line);
-        return msgs.map((m) =>
-            `<div class="message ${m.severity}">
-                <h1 title="${m.file_name}:${m.pos_line}:${m.pos_col}">
-                  <a href="${encodeURI('command:lean.revealPosition?' + JSON.stringify([Uri.file(m.file_name), m.pos_line - 1, m.pos_col]))}">
-                    ${basename(m.file_name)}:${m.pos_line}:${m.pos_col}: ${m.severity} ${m.caption}
-                  </a>
-                </h1>
-                <pre>${m.text}</pre>
-            </div>`).join("\n");
     }
 
-    private rerender() {
-        if (this.paused) return;
-        this.render();
-        this.changedEmitter.fire();
-    }
+	private getMediaPath(mediaFile: string): string {
+		return Uri.file(this.context.asAbsolutePath(join('media', mediaFile))).toString();
+	}
 
-    private togglePause() {
-        this.paused = !this.paused;
-        this.render();
-        this.changedEmitter.fire();
-    }
+    private render() {
+        return `<!DOCTYPE html>
+            <html>
+            <head> 
+            <meta http-equiv="Content-type" content="text/html;charset=UTF-8">
+            <style>
+            div
```

Comment:
When I used a `<link src="stylesheet.css" ...>` the CSS sporadically wasn't applied when waking over the text. Embedding it into the HTML helped fixing this. But you are right, the extra file was very helpful for me. I will try to load it on plugin start, and add a configuration option to add extra CSS snippets.